src/middlewared/middlewared/main.py

```diff
@@ -229,15 +211,13 @@ async def subscribe(self, ident, name):
     async def unsubscribe(self, ident):
         if ident in self.__subscribed:
             self.__subscribed.pop(ident)
-        elif ident in self.__event_sources:
-            event_source = self.__event_sources[ident]['event_source']
-            await self.middleware.run_in_thread(event_source.cancel)
-            self.__event_sources.pop(ident)
+        elif self.session_id + ident in self.middleware.event_source_manager.idents:
```

Comment:
Do you think we should perhaps have a method which determines the unique identifier so that we would not have to do `self.session_id + ident` everywhere ?